r/R/dplyr.R

```diff
@@ -219,6 +219,31 @@ tail.arrow_dplyr_query <- function(x, n = 6L, ...) {
   x
 }
 
+#' Show the details of an Arrow Execution Plan
+#'
+#' This is a function which gives more details about the Execution Plan (`ExecPlan`)
+#' of an `arrow_dplyr_query` object. It is similar to `dplyr::explain()`.
+#'
+#' @param x an `arrow_dplyr_query` to print the `ExecPlan` for.
+#'
+#' @return The argument, invisibly.
+#' @export
+#'
+#' @examplesIf arrow_with_dataset() & requireNamespace("dplyr", quietly = TRUE)
+#' library(dplyr)
+#' mtcars %>%
+#'   arrow_table() %>%
+#'   filter(mpg > 20) %>%
+#'   mutate(x = gear/carb) %>%
+#'   show_exec_plan()
+show_exec_plan <- function(x) {
+  adq <- as_adq(x)
```

Comment:
No. Removed.

Reply:
Yes, they are. I thought it might be useful not to error if someone is building a pipeline and the first step is:
```r
mtcars %>% arrow_table() %>% show_exec_plan()
```

Reply:
The tests weren't covering the case. 

If we do not coerce to `adq`, we get the following error:
``` r
mtcars %>%
    arrow_table() %>%
    show_exec_plan()
#> Error: Invalid: Invalid column index to remove field.
#> /Users/dragos/Documents/arrow/cpp/src/arrow/table.cc:110 schema_->RemoveField(i)
```

Reply:
=> so I added a unit test